lib/puppet/provider/package/dpkg.rb

```diff
@@ -113,6 +120,20 @@ def query
 
     # list out our specific package
     begin
+      if @resource.allow_virtual?
+        output = dpkgquery(
+          "-W",
+          "--showformat",
+          self.class::DPKG_QUERY_PROVIDES_FORMAT_STRING
+          #the regex searches for the resource[:name] in the dpkquery result in which the Provides field is also available
+          #it will search for the packages only in the brackets ex: [rubygems]
+        ).lines.find {|package| package.match(/[\[ ](#{Regexp.escape(@resource[:name])})[\],]/)}
```

Comment:
The comma makes me think there be multiple package names inside the provides, eg `[foo, foo-dev, bar]`? Could you add a test for that?

Reply:
I think this is already tested using this query_output
https://github.com/puppetlabs/puppet/pull/7701/files#diff-68fbdeeded08eb112e8ce141c005d7beR96